autogen.sh

```diff
@@ -1,5 +1,23 @@
 #!/bin/sh
 
+set -e
+
+# Commands this script needs
+needed='rm mkdir autoreconf echo'
+if ! type $needed >/dev/null
```

Comment:
`type` will output to stderr when not found, so use `if ! type $needed &>/dev/null` here and on line 11.